From 5e89bbc79ebc920e42b469248c4b3e5244dd548f Mon Sep 17 00:00:00 2001
Message-Id: <5e89bbc79ebc920e42b469248c4b3e5244dd548f@dist-git>
From: Eric Blake <eblake@redhat.com>
Date: Mon, 19 Aug 2013 16:32:31 -0600
Subject: [PATCH] virt-xml-validate: add missing schemas

https://bugzilla.redhat.com/show_bug.cgi?id=1006699

We were failing to autoprobe which schema to use for several
top-level XML elements.

* tools/virt-xml-validate.in (TYPE): Recognize <domainsnapshot>,
<filter>, and <secret>.

Signed-off-by: Eric Blake <eblake@redhat.com>
(cherry picked from commit ab4304b7e0b90a18c5ec528028d0f1a6d1317c8c)
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 tools/virt-xml-validate.in | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/virt-xml-validate.in b/tools/virt-xml-validate.in
index 27ad59e..36bc6f2 100644
--- a/tools/virt-xml-validate.in
+++ b/tools/virt-xml-validate.in
@@ -33,6 +33,9 @@ fi
 if [ -z "$TYPE" ]; then
   ROOT=`xmllint --stream --debug "$XMLFILE" 2>/dev/null | grep "^0 1 " | awk '{ print $3 }'`
   case "$ROOT" in
+     *domainsnapshot*) # Must come first, since *domain* is a substring
+        TYPE="domainsnapshot"
+        ;;
      *domain*)
         TYPE="domain"
         ;;
@@ -51,6 +54,12 @@ if [ -z "$TYPE" ]; then
      *device*)
         TYPE="nodedev"
         ;;
+     *filter*)
+        TYPE="nwfilter"
+        ;;
+     *secret*)
+        TYPE="secret"
+        ;;
      *)
        echo "$0: cannot determine schema type for $XMLFILE"
        exit 3
-- 
1.9.2

